File: rockPaperScissors.c
#include <stdio.h>
#include <time.h>
#include <math.h>
#include <stdlib.h>
#include <ctype.h>

void results(int points, char* username);
int playGame(char userMove, char computerMove);
char userPlay();
char computerPlay();
char * createUsername();

int main()
{
	
	printf("Welcome to rock, paper, scissors with C!\n\n");

	char* username = createUsername();
	char userMove = userPlay();
	char computerMove = computerPlay();

	int points = playGame(userMove, computerMove);
	results(points, username);
	
	return 0;
}

int playGame(char userMove, char computerMove) 
{
	switch (userMove) {
	case 'r':
		if (computerMove == 's') {
			return 1;
		} else if (computerMove == 'p') {
			return -1;
		} else {
			return 0;
		}
	case 'p':
		if (computerMove == 'r') {
			return 1;
		} else if (computerMove == 's') {
			return -1;
		} else {
			return 0;
		}
	case 's':
		if (computerMove == 'p') {
			return 1;
		} else if (computerMove == 'r') {
			return -1;
		} else {
			return 0;
		}
	}
	return 2;
}

void results(int points, char* username) 
{
	if (points == 0) {
		printf("\nAlas it was a draw\n\n");
	} else if (points == 1) {
		printf("\nWhoop whoop you won %s!\n\n", username);
	} else if (points == -1) {
		printf("\nBoo you lost %s :(\n\n", username);
	} else {
		printf("\nThere seems to have been a mistake somewhere\n\n");
	}
}

char userPlay() 
{
	static char userChoice;
	printf("\nPlease enter one of the following options:  \n");
	printf("R for rock, P for paper, S for scissors and E for exit:  ");
	scanf("\n%c", &userChoice);
	
	if (toupper(userChoice) == 'R') {
		return 'r';
	} else if (toupper(userChoice) == 'P') {
		return 'p';
	} else if (toupper(userChoice) == 'S') {
		return 's';
	} else if (toupper(userChoice) == 'E') {
		printf("\nGoodbye...\n");
		exit(0);
	} else {
		printf("\nyou didn't enter any of the options provided, im very sad :(\n\n");
		return userPlay();
	}
	return 'h';
}
char computerPlay() 
{
	//this creates a random number between 1 and 3
	srand(time(NULL));
	int randomNumber = rand() % 3 + 1;
	
	char compMove;
	if (randomNumber == 1) {
		printf("\nThe computer played rock\n");
		compMove = 'r';
	} else if (randomNumber == 2) {
		printf("\nThe computer played paper\n");
		compMove = 'p';
	} else if (randomNumber == 3) {
		printf("\nThe computer played scissors\n");
		compMove = 's';
	} else {
		compMove = 'e';
		printf("\nerror has occurred!\n");
	}
	return compMove;
}

char * createUsername()
{
	// probably better ways to do this than using static
	// look this up :)
	static char username[20];
	printf("Please enter your username: \n");
	scanf("%s", username);
	
	return username;
	
}

File: rockPaperScissors.py
import random

def results(points, username):
	if points == 0:
		print("\nAlas it was a draw\n")
	elif points == 1:
		print("\nWhoop whoop you won " +  username + "!\n")
	elif points == -1:
		print("\nBoo you lost " + username + ":(\n")
	else:
		printf("\nThere seems to have been a mistake somewhere\n")


def playGame(userMove, computerMove):

	match userMove:
		case 'r':
			if computerMove == 's':
				return 1
			elif computerMove == 'p':
				return -1
			else:
				return 0;
		case 'p':
			if computerMove == 'r':
				return 1
			elif computerMove == 's':
				return -1
			else:
				return 0
		case 's':
			if computerMove == 'p':
				return 1
			elif computerMove == 'r':
				return -1
			else:
				return 0
	

def computerPlay():
	randomNumber = random.randint(1,3)
	
	if randomNumber == 1:
		print("The computer played rock")
		compMove = 'r'
	elif randomNumber == 2:
		print("The computer played paper")
		compMove = 'p'
	elif randomNumber == 3:
		print("The computer played scissors")
		compMove = 's'
	else:
		compMove = 'e'
		print("error has occurred!")
	return compMove;

def userPlay():

	print("\nPlease enter one of the following options:")
	userChoice = input("R for rock, P for paper, S for scissors and E for exit:  ")
	if userChoice.upper() == 'R':
		return 'r'
	elif userChoice.upper() == 'P':
		return 'p'
	elif userChoice.upper() == 'S':
		return 's'
	elif userChoice.upper() == 'E':
		print("\nGoodbye...")
		quit()
	else:
		print("you didn't enter any of the options provided, im very sad :(")
		return userPlay()


def createUsername():
	
	username = input("Please enter your username :\n")
	return username
	

if __name__ == "__main__":
	print("Welcome to rock, paper, scissors with Python!\n");

	username = createUsername()
	userMove = userPlay();
	computerMove = computerPlay();
	
	points = playGame(userMove, computerMove);
	results(points, username);



	
